docs/gno-infrastructure/setting-up-a-faucet.md

```diff
@@ -0,0 +1,263 @@
+
+Below are errors you may run into when setting up or using the faucet.
+
+### During setup
+
+When setting up the faucet, you can run into the following errors:
+- If the faucet listen address is invalid or is taken - `invalid listen address`
+- If the chain ID the faucet connects to is invalid - `invalid chain ID`
+- If the send amount defined is invalid - `invalid send amount`
+- If the mnemonic used for the faucet is invalid - `invalid mnemonic`
+- If the number of accounts to derive from the mnemonic is less than zero -
+`invalid number of faucet accounts`
+
+### During requests
+
+When requesting a drip from the faucet, you can face the following errors:
+- If the address provided is empty or has an invalid checksum - `invalid beneficiary address`
+- If the amount requested is empty, not in the `<amount>ugnot` format, or is larger
+than `send_amount` defined in the faucet configuration
+
+## `gnofaucet`
+
+[`gnofaucet`](https://github.com/gnolang/gno/tree/master/contribs/gnofaucet) 
+is an extended version of the base faucet. It includes two main security features 
+as an extension to the base version:
+- IP Throttling using [X-Forwarded-For](https://en.wikipedia.org/wiki/X-Forwarded-For)
+- Integrated captcha check using [reCaptcha V2](https://developers.google.com/recaptcha/docs/display)
+
+Let's see how we can set it up.
+
+## Cloning the repo
+
+To get started with setting up the `gnofaucet`, visit the
+[Gno monorepo](https://github.com/gnolang/gno) and clone it:
+
+```bash
+git clone git@github.com:gnolang/gno.git
+```
+
+After cloning, go into the `contribs/gnofaucet/` folder, where you will find
+the `gnofaucet` implementation. Then, run the following `make` command to build 
+out the binary:
+
+```bash
+make build
+```
+
+## Starting `gnofaucet`
+
+To start the faucet, we can use the `serve` subcommand. In the case of `gnofaucet`,
+we have a more minimalistic set of flags than in the base faucet, and we do not
+have access to a configuration file. Instead, we configure everything via flags.
+
+Running the following help command will print out the available flags:
+
+```bash
+./build gnofaucet serve --help
+```
+
+Below is the output of the command:
+
+```bash
+-captcha-secret ...             recaptcha secret key (if empty, captcha are disabled)
+-chain-id ...                   the chain ID associated with the remote Gno chain
+-is-behind-proxy=false          use X-Forwarded-For IP for throttling
+-listen-address 127.0.0.1:5050  the faucet server listen address
+-max-send-amount 10000000ugnot  the static max send amount (native currency)
+-mnemonic ...                   the mnemonic for faucet keys
+-num-accounts 1                 the number of faucet accounts, based on the mnemonic
+-remote http://127.0.0.1:26657  remote node URL
+```
+
+In this case, only two new flags exist:
+- `captcha-secret` - where you can provide your reCaptcha v2 secret key
+- `is-behind-proxy` - which enables or disables the IP throttling functionality.
```

Comment:
Can you please standardize the formatting in this doc?
Some points have `.`, some don't